rmw_zenoh_cpp/src/detail/rmw_client_data.cpp

```diff
@@ -277,13 +273,6 @@ liveliness::TopicInfo ClientData::topic_info() const
   return entity_->topic_info().value();
 }
 
-///=============================================================================
-bool ClientData::liveliness_is_valid() const
-{
-  std::lock_guard<std::recursive_mutex> lock(mutex_);
-  return zc_liveliness_token_check(&token_);
-}
```

Comment:
I understand that these `z_check` functions have been made internal but having a method to check whether the liveliness token is still valid is useful. How about we simply return `!is_shutdown_` here since we unregister the token only after the entity is shutdown. Then keep the function call in `rmw_zenoh.cpp`.

Reply:
We only check `liveliness_is_valid` in _rmw_zenoh_cpp/src/rmw_zenoh.cpp_. Let me change it with `is_shutdown()` there.

rolling branch
```txt
rmw_zenoh_cpp/src/rmw_zenoh.cpp
754:  if (!pub_data->liveliness_is_valid()) {

rmw_zenoh_cpp/src/detail/rmw_service_data.hpp
63:  bool liveliness_is_valid() const;

rmw_zenoh_cpp/src/detail/rmw_service_data.cpp
242:bool ServiceData::liveliness_is_valid() const

rmw_zenoh_cpp/src/detail/rmw_publisher_data.cpp
431:bool PublisherData::liveliness_is_valid() const

rmw_zenoh_cpp/src/detail/rmw_publisher_data.hpp
74:  bool liveliness_is_valid() const;

rmw_zenoh_cpp/src/detail/rmw_subscription_data.hpp
90:  bool liveliness_is_valid() const;

rmw_zenoh_cpp/src/detail/rmw_client_data.cpp
281:bool ClientData::liveliness_is_valid() const

rmw_zenoh_cpp/src/detail/rmw_subscription_data.cpp
389:bool SubscriptionData::liveliness_is_valid() const

rmw_zenoh_cpp/src/detail/rmw_client_data.hpp
64:  bool liveliness_is_valid() const;
```